internal/jujuapi/jimm.go

```diff
@@ -41,7 +41,8 @@ func init() {
 		addCloudToControllerMethod := rpc.Method(r.AddCloudToController)
 		removeCloudFromControllerMethod := rpc.Method(r.RemoveCloudFromController)
 		addGroupMethod := rpc.Method(r.AddGroup)
-		getGroupMethod := rpc.Method(r.GetGroup)
```

Comment:
This is not backwards compatible and would break existing Terraform provider code that has landed. We could do work to go and remove it from their codebase before is shipped, but that is not how we should build.

I think in the case of the public API, we can keep it as just `GetGroup` and we can add a field to `GetGroupRequest` so it has a UUID and a name, then in the jujuapi handler, if both are set, return an error.

OR we can keep `GetGroup` which queries by UUID and add `GetGroupByName` which queries by name.